docs/designs/termination.md

```diff
@@ -0,0 +1,130 @@
+# Karpenter Graceful Node Termination
+
+## Overview
+Karpenter's scale down implementation is currently a proof of concept. The reallocation controller implements two actions. First, nodes are elected for termination when there aren't any pods scheduled to them. Second, Node are cordoned and drained and deleted. Node termination follows cordon and drain [best practices](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/). 
+
+This design explores robustness improvements to the termination process and proposes the separation of this logic into a new termination controller. 
```

Comment:
You can probably remove "robustness" here, the design is just improving up the termination process.  Just want to make the doc as clear and concise as possible :)

Reply:
The logic separation into a new termination controller will still live in the same binary right?  If so, can you make that explicit so it doesn't seem like this new thing is going to be a stand-alone component.

Reply:
Good callouts, improvements are improvements :)

The new termination controller will live in the same binary, but won't require a provisioner to be created for it to work. In that sense, it'll be a stand-alone component, but still a part of Karpenter. Thanks!